torch/onnx/_internal/exporter/_torchlib/ops/nn.py

```diff
@@ -119,19 +119,19 @@ def aten_scaled_dot_product_attention_23(
                 "SDPA (MHA) requires q_num_heads = kv_num_heads"
             )
 
-        # NOTE: There was extended discussion on whether the num_heads attributes (q_num_heads/kv_num_heads)
-        # should be set as ONNX attributes or inferred from the tensor shape. In ONNX, num_heads is needed
-        # for 3D attention inputs (shape: [B, S, N*H]), but not for 4D ([B, N, S, H]), which is the only
-        # input accepted by this exporter. Thus, the attribute is not strictly necessary here, but adding it
-        # may ease future optimization or conversion to 3D formats (e.g., GQA ops)
+        # NOTE: num_heads attributes (q_num_heads/kv_num_heads) are not mandatory for 4D.
```

Comment:
nit: may be clearer if we change "are not mandatory for 4D" to "should NOT be specified for 4D (even though the ONNX op documentation does not make this clear)"